cooperator/tests/test_cooperator.py

```diff
@@ -2,58 +2,69 @@
 #   Robin Keunen <robin@coopiteasy.be>
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
-import datetime
+from datetime import date, datetime, timedelta
 
 from odoo.exceptions import AccessError
 from odoo.fields import Date
+from odoo.tests.common import SavepointCase, users
 
-from .test_base import CooperatorBaseCase
+from .cooperator_test_mixin import CooperatorTestMixin
 
 
-class CooperatorCase(CooperatorBaseCase):
-    def setUp(self):
-        super().setUp()
-
-        self.request = self.browse_ref("cooperator.subscription_request_1_demo")
-        self.bank_journal_euro = self.env["account.journal"].create(
+class CooperatorCase(SavepointCase, CooperatorTestMixin):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.set_up_cooperator_test_data()
+        cls.bank_journal = cls.env["account.journal"].create(
             {"name": "Bank", "type": "bank", "code": "BNK67"}
         )
-        self.payment_method_manual_in = self.env.ref(
-            "account.account_payment_method_manual_in"
+        cls.payment_method = cls.env.ref("account.account_payment_method_manual_in")
+        cls.share_line = cls.env["share.line"].create(
+            {
+                "share_product_id": cls.share_x.id,
+                "share_number": 2,
+                "share_unit_price": 50,
+                "partner_id": cls.demo_partner.id,
+                "effective_date": datetime.now() - timedelta(days=120),
+            }
         )
 
+    @users("user-cooperator")
     def test_put_on_waiting_list(self):
-        self.as_cooperator_user()
-        self.request.put_on_waiting_list()
-        self.assertEqual(self.request.state, "waiting")
+        self.subscription_request_1.put_on_waiting_list()
+        self.assertEqual(self.subscription_request_1.state, "waiting")
 
+    @users("user-cooperator")
     def test_validate_subscription_request(self):
+        self.subscription_request_1.validate_subscription_request()
+
+        self.assertEqual(self.subscription_request_1.state, "done")
+        self.assertTrue(self.subscription_request_1.partner_id)
+        self.assertTrue(self.subscription_request_1.partner_id.coop_candidate)
+        self.assertFalse(self.subscription_request_1.partner_id.member)
+        self.assertEqual(self.subscription_request_1.type, "new")
+        self.assertTrue(len(self.subscription_request_1.capital_release_request) >= 1)
+        self.assertEqual(
+            self.subscription_request_1.capital_release_request.state, "open"
+        )
+        self.assertTrue(self.subscription_request_1.capital_release_request.sent)
+
+    @users("user-cooperator")
+    def test_register_payment_for_capital_release(self):
         self.as_cooperator_user()
```

Comment:
a leftover? should produce and error, but it does not happen since this test is not run (because of the duplicate method name).